spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/connection/TcpNioConnection.java

```diff
@@ -662,6 +662,7 @@ protected synchronized void doWrite(ByteBuffer buffer) throws IOException {
 				TcpNioConnection.this.socketChannel.write(buffer);
 				remaining = buffer.remaining();
 			}
+			selector.close();
```

Comment:
This doesn't look OK to me.
We have a logic in this method like:
```
if (this.selector == null) {
				this.selector = Selector.open();
```
This `close()`  will just close the selector, but would leave `this.selector` as not null.
Probably better to close it in the `ChannelOutputStream.close()` impl...

Reply:
I added ```selector = null``` hannelOutputStream.close() is not doing the trick after the close pipes/fd are left open and eventually the process ending up Out Of file descriptors.